libdrgn/python/error.c

```diff
@@ -123,6 +125,13 @@ DRGNPY_PUBLIC void *set_drgn_error(struct drgn_error *err)
 		PyErr_SetString(PyExc_LookupError, err->message);
 		break;
 	case DRGN_ERROR_FAULT:
+		if (err->address) {
+			address = PyLong_FromUnsignedLongLong(*err->address);
+			if (!address)
+				PyErr_NoMemory();
+		}
+		// XXX: retval?
+		(void) PyObject_SetAttrString(FaultError, "address", address);
```

Comment:
This is setting the `address` attribute on the `FaultError` class itself. Ideally we would have the equivalent of the following:

```python
class FaultError(Exception):
    def __init__(self, message, address):
        self.message = message
        self.address = address

    def __str__(self):
        return f'{self.message}: {self.address:#x}'
```

Then here we could use something like
```c
PyObject *exc = PyObject_CallFunction(&FaultError_type, "sK", err->message, err->address);
PyErr_SetObject(&FaultError_type, exc);
```

This means that `FaultError_type` needs to be defined manually rather than with `PyErr_NewExceptionWithDoc`. The best example I could find was [here](https://github.com/python/cpython/blob/47be7d0108b4021ede111dbd15a095c725be46b7/Modules/_ctypes/_ctypes.c#L5515).